### DISCUSSION TITLE
Creating floating UI components like Google Maps AR navigation example

### DISCUSSION THREAD

**Discussion by bounty-torn** [google/filament#4095] [Q&A] [answered]
I am a noob developer that is wondering how the Google Maps AR navigation UI components were created (the "Continue this way" components), specifically text elements. Are they just 3D objects (with baked-in text) that are part of the scene rendered by Filament? How do we create text-based UI elements? Any help (even seemingly basic, elementary help) would be greatly appreciated, as I don't even know where to begin.
[Google Maps AR Navigation](https://github.com/google/filament/raw/main/docs/images/samples/app_gmm_ar_nav.jpg)

**Comment by prideout** [comment, marked as answer]
Filament does not have text-based UI facility, in your screenshot each signpost is probably a 4-vertex mesh with a texture.  To generate a text texture, there are several approaches you could take.  For example you could use `stb_truetype.h`.  Alternatively you could draw text into an Android Bitmap in the Java layer, then convert that into a texture using our `TextureHelper` class.

**Reply by romainguy** [reply]
In this particular screenshot the arrows are a 3D model and the text cards are Android Views (from the UI toolkit) rendered as textures inside Filament.
